src/config.ts

```diff
@@ -14,3 +14,146 @@ export const dfinity_bridge = process.env.DFINITY_BRIDGE;
 export const config_scan: TExplorerConfig =
   process.env.EVM_WHITELIST_CONFIG &&
   JSON.parse(process.env.EVM_WHITELIST_CONFIG);
+export interface ChainConfig {
+  name: string;
+  chainFactory: any;
+  nonce: string;
+}
+interface Config {
+  web3: ChainConfig[];
+}
+
+export const getChainFactory = async (chain: string) => {
+  const con = await ChainFactoryConfigs.MainNet();
+  return con[chain];
+};
+
+const chainData: Config = {
+  web3: [
+    {
+      name: 'ABEYCHAIN',
+      chainFactory: getChainFactory('abeyChainParams'),
```

Comment:
Add args for getChainFactory() in a const. Even better would be to narrow down typings for chain on line 26. For example

type TChainParams = "abeyChainParams" | "auroraParams" ...

export const getChainFactory = async (chain: TChainParams) => { ... }